driver/task.go

```diff
@@ -27,6 +34,7 @@ type Task struct {
 	VarFiles        []string
 	Version         string
 	UserDefinedMeta map[string]map[string]string
+	BufferPeriod    *BufferPeriod
```

Comment:
I don't think this needs to be a pointer. For changes to the Task, it is already a pointer. I don't see any mutations of BufferPeriod in place and I don't see any cases where the nil case is needed (checking for unset doesn't seem required over 0-case) in either this PR now #185.

Reply:
I suggest this as the semantics for 0-values is much nicer (no panics) and values allow for better performance (can be allocated on the stack, so less GC).

Reply:
Yea, there isn't any mutations of BufferPeriod. I made BufferPeriod a pointer so that the nil case would be meaningful: nil = buffer_period is disabled.

The alternative would be instead to have an enabled field on the BufferPeriod. This was a little less intuitive to me from the backend since disabled BufferPeriod should not have mix/max values. However, this would be more in line with the configuration structure 🤔 .

What do you think? You mention a lot of other positives for not having a pointer. Is your preference still for not having a pointer? Happy to change, just want to double-check.

Reply:
If nil is used as an unset/disabled value where the 0 value won't work, then it makes sense. I probably missed or misinterpreted where that was done. Using an enabled field is an option but I don't think it is a slam dunk improvement as it has downsides as well.

TLDR; it is being used to represent 'disabled' and no change is necessary.

Reply:
Gotcha! Thanks for your input and helpful to hear your train of thought. Let me add a comment to clarify that the nil value is 'disabled' since it's not not very apparent.